test/test_download_file.py

```diff
@@ -222,3 +222,28 @@ def test_the_user_specifies_a_custom_slice_size_different_to_before(mock_data_cl
     assert not os.path.exists(extra_slice)
     assert "Deleting the leftover" in caplog.text
 
+
+def test_slice_file_is_reused(mock_data_client, mock_data_server, random_binary_file, caplog):
```

Comment:
Love seeing new tests added! Should there be a change somewhere that calls this new test function?

Reply:
Good catch and good question! But no, there's no need for calling this function, because the test framework [will call automatically](https://docs.pytest.org/en/6.2.x/getting-started.html#group-multiple-tests-in-a-class) the test functions (=those functions which start with test_), when the test framework is started.

Reply:
Ah, thanks for the explanation!